File: Forge/src/main/java/hantonik/atomic/core/inventory/slot/AtomicSingleSlot.java
package hantonik.atomic.core.inventory.slot;

import net.minecraft.world.item.ItemStack;
import net.minecraftforge.items.IItemHandler;
import net.minecraftforge.items.SlotItemHandler;

public class AtomicSingleSlot extends SlotItemHandler {
    public AtomicSingleSlot(IItemHandler itemHandler, int index, int xPosition, int yPosition) {
        super(itemHandler, index, xPosition, yPosition);
    }

    @Override
    public int getMaxStackSize(ItemStack stack) {
        return 1;
    }

    @Override
    public int getMaxStackSize() {
        return 1;
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/inventory/slot/AtomicSingleSlot.kt
package hantonik.atomic.core.inventory.slot

import net.minecraft.world.item.ItemStack
import net.minecraftforge.items.IItemHandler
import net.minecraftforge.items.SlotItemHandler

open class AtomicSingleSlot(inventory: IItemHandler, id: Int, x: Int, y: Int) : SlotItemHandler(inventory, id, x, y) {
    override fun getMaxStackSize(stack: ItemStack) = 1

    override fun getMaxStackSize() = 1
}